### PR TITLE
Add performance tests for #2925

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Adds performance and tck tests confirming #2925 

Note that this PR does **not** fix the performance issue

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
| 🟥 aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 20101954 | 3870 | 3979 | 2008534 |
| 🟥 aggregations.AggregationWhereWithinNestedConnections | 22116987 | 20101954 | 3719 | 3597 | 2008534 |
| 🟥 aggregations.NestedCountFromMovieToActors | 9734 | 8694 | 58 | 106 | 2174 |
| 🟥 aggregations.NestedCountFromActorsToMovie | 9937 | 8900 | 79 | 206 | 2174 |
| 🟥 batch-create.BatchCreate | 4200 | 3600 | 166 | 139 | 600 |
| 🟥 connect.createAndConnect | 14424 | 12419 | 278 | 300 | 3003 |
| 🟥 create.SimpleMutation | 7 | 6 | 78 | 48 | 1 |
| 🟥 cypher-directive.TopLevelCypherDirective | 116092 | 99525 | 960 | 1118 | 12241 |
| 🟥 cypher-directive.TopLevelCypherDirectiveWithColumnName | 147609 | 127994 | 410 | 462 | 12241 |
| 🟥 delete.NestedDeleteInUpdate | 22544 | 17123 | 217 | 303 | 2040 |
| 🟦 2925.SingleRelationshipFilter | 6468 | N/A | 45 | N/A | 1040 |
| 🟥 query.QueryWhere | 9694 | 8658 | 42 | 68 | 2161 |
| 🟥 query.SimpleQueryWithNestedWhere | 9872 | 8829 | 64 | 99 | 2161 |
| 🟥 query.OrFilterOnRelationships | 42626 | 36060 | 196 | 384 | 1648 |
| 🟥 query.OrFilterOnRelationshipsAndNested | 36620 | 30122 | 253 | 517 | 1648 |
| 🟥 query.QueryWithNestedIn | 13747 | 12239 | 77 | 72 | 1334 |
| 🟥 query.NestedConnectionWhere | 9834 | 8794 | 65 | 79 | 2174 |
| 🟥 query.DeeplyNestedConnectionWhere | 9991 | 8898 | 112 | 146 | 2174 |


🟥 - Performance worsened (dbHits)
🟩 - Performance improved (dbHits)
🟦 - New test
        

<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 69 | 60 | 1134 |
|  aggregations.NestedAggregation | 14551 | 13514 | 109 | 150 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 10942 | 85 | 136 | 2174 |
| 🟥 aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 20101954 | 3870 | 3979 | 2008534 |
| 🟥 aggregations.AggregationWhereWithinNestedConnections | 22116987 | 20101954 | 3719 | 3597 | 2008534 |
| 🟥 aggregations.NestedCountFromMovieToActors | 9734 | 8694 | 58 | 106 | 2174 |
| 🟥 aggregations.NestedCountFromActorsToMovie | 9937 | 8900 | 79 | 206 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 12062316 | 5548 | 6032 | 2008534 |
| 🟥 batch-create.BatchCreate | 4200 | 3600 | 166 | 139 | 600 |
| 🟥 connect.createAndConnect | 14424 | 12419 | 278 | 300 | 3003 |
|  connections.Connection | 14082 | 13042 | 109 | 192 | 2174 |
|  connections.NestedConnection | 41462 | 38231 | 144 | 166 | 4516 |
|  connections.ConnectionWithSort | 2282 | 2277 | 74 | 117 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2282 | 2277 | 300 | 494 | 1040 |
| 🟥 create.SimpleMutation | 7 | 6 | 78 | 48 | 1 |
| 🟥 cypher-directive.TopLevelCypherDirective | 116092 | 99525 | 960 | 1118 | 12241 |
| 🟥 cypher-directive.TopLevelCypherDirectiveWithColumnName | 147609 | 127994 | 410 | 462 | 12241 |
|  delete.SimpleDelete | 19401 | 18361 | 429 | 404 | 1040 |
| 🟥 delete.NestedDeleteInUpdate | 22544 | 17123 | 217 | 303 | 2040 |
| 🟦 2925.SingleRelationshipFilter | 6468 | N/A | 45 | N/A | 1040 |
|  query.SimpleQuery | 15160 | 14120 | 110 | 153 | 2174 |
| 🟥 query.QueryWhere | 9694 | 8658 | 42 | 68 | 2161 |
| 🟥 query.SimpleQueryWithNestedWhere | 9872 | 8829 | 64 | 99 | 2161 |
|  query.Nested | 10092037 | 10084290 | 11928 | 11926 | 2008534 |
|  query.NestedWithFilter | 40401 | 37172 | 145 | 157 | 4511 |
| 🟥 query.OrFilterOnRelationships | 42626 | 36060 | 196 | 384 | 1648 |
| 🟥 query.OrFilterOnRelationshipsAndNested | 36620 | 30122 | 253 | 517 | 1648 |
| 🟥 query.QueryWithNestedIn | 13747 | 12239 | 77 | 72 | 1334 |
| 🟥 query.NestedConnectionWhere | 9834 | 8794 | 65 | 79 | 2174 |
| 🟥 query.DeeplyNestedConnectionWhere | 9991 | 8898 | 112 | 146 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6343 | 6251 | 231 | 214 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3024400 | 3021189 | 822 | 832 | 1003942 |
|  query.Fulltext | 96 | 96 | 33 | 38 | 16 |
|  query.FulltextWithNestedQuery | 587 | 571 | 51 | 60 | 84 |
|  sorting.SortMultipleTypes | 2513 | 2493 | 76 | 89 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1477 | 1406 | 364 | 426 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 13042 | 72 | 107 | 2174 |
|  sorting.SortDeeplyNestedFields | 42196 | 38965 | 155 | 162 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 198 | 344 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 75 | 153 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 69 | 144 | 35 |
|  unions.NestedUnion | 410637 | 398553 | 346 | 473 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 372527 | 371 | 399 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 84 | 68 | 1040 |

</details>
        
Old Schema Generation: 46.575s
Schema Generation: 43.137s

**Comment by angrykoala** [comment]
Note that these performance tests are slower due to new data added to the set

File: packages/graphql/tests/tck/issues/2925.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
+            "MATCH (this:\`User\`)
+            WHERE single(this0 IN [(this)-[:HAS_GROUP]->(this0:\`Group\`) WHERE this0.name IN $param0 | 1] WHERE true)
+            RETURN this { .name } AS this"
+        `);
+
+        // MATCH (this:\`User\`)-[:HAS_GROUP]->(this0:\`Group\`)
+        // WHERE this0.name IN $param0
+        // RETURN this { .title } AS this
```
Comment (darrellwarde) [inline packages/graphql/tests/tck/issues/2925.test.ts:64]: Do we need these comments here?

**Review by darrellwarde — APPROVED** [review]
LGTM!

**Pull request merged**